### PR TITLE
Update git-version-bump.rb

### PR DESCRIPTION
**Pull request by p64**
"bundle exec git-version-bump -n p" seems to just hang for me. This fixed it for me.

Maybe I'm doing something wrong with the tool.  But VIM would not fork until I turned this into spawn.  Tried on fedora 37 and Rocky9.   Assuming spawn retains the tty vs Open3.capture2e which seems to be hijacking it.

I don't know if others have this problem or use this.

Gem really helped me with tagging!

### CODE REVIEW

File: lib/git-version-bump.rb
Comment (mpalmer) [inline lib/git-version-bump.rb:249]: Since we're not capturing the output here, the `out` variable won't be initialized.
```suggestion
                        raise CommandFailure.new("Failed while #{desc}", "(output not captured)", status.exitstatus)
```

File: lib/git-version-bump.rb
Comment (mpalmer) [inline lib/git-version-bump.rb:232]: Given how much of this method's body is shared with `run_command`, I'm inclined to add a kwarg (with a default) to `run_command` to specify "don't capture the command output", rather than duplicating the method's code.  It's a close-run thing, because it's complexity either way -- a zillion kwargs is at least as much of a mess as near-duplicate methods, but in this case I *think* the kwarg is better.

**Review by mpalmer — CHANGES_REQUESTED** [review]
Huh, that's really interesting.  I totally see why an editor would get stroppy about having its output captured, which makes me wonder how this has ever worked for *anyone* (myself included).  Possibly a terminal/OS/termcap/editor/phase-of-the-moon difference?

In any event, I think your solution is the right one.  I've got a couple of quibbles about your exact implementation, which if you'd like to have a crack at addressing, I'll be happy to merge after that.  If you don't have the time, energy, or inclination, don't sweat it -- just let me know and I'll make the fix.

**Comment by p64** [comment]
Thanks for reaching out.  My only goals are:

1. it works, which I'm with you, I can't see how that would ever work for anyone.  But I'm not above not understanding something or being wrong about something.  Maybe I was doing something wrong.
2. That I don't have to keep merging the patch as bundle update happens.

That said I tried to adhere to the style, and my best idea was to dupe run.  At first I called it run_command2 ... which made me feel extremely dirty.  Maybe it could be something like run_command_spawn or anything.  I just didn't want "2" and I figured it was retaining the tty, but maybe just spawn_no_out or something.  Your call.  Let me know what if anything I need to do to help.  Happy to do whatever is needed to achieve 1&2 above. :)

File: lib/git-version-bump.rb
Comment (p64) [reply lib/git-version-bump.rb:232]: I agree with that, was trying to "when in rome..." :) But your call.  Happy to help in any way.  I really dislike copy / paste.  The other thing I was thinking was maybe it should return the status.exitcode, but we weren't checking the return code so I didn't do that.  But it would be cleaner to return something that might be helpful, in my opinion.

Comment (mpalmer) [reply lib/git-version-bump.rb:232]: I decided to raise an exception when a command fails, rather than return the exit code, because a failing command is (usually) exceptional.

I think what feels best is to:

1. change the `run_command` method to have the signature of `run_command(cmd, desc, capture_output: true)';
2. adjust the method code to match;
3. when output isn't being captured:
  1. the `out` arg to the exception can be `(output not captured)`, as you've done in your commit;
  2. the return value is `nil`;
  3. `spawn` is called rather than `capture2e`.

**Comment by mpalmer** [comment]
Yeeeeaah, calling it `run_command2` would have gotten a :raised_eyebrow: comment... :grinning:  I don't think you're doing anything wrong, as I said before I totally can't see how it could be working as currently written.  I'm going with "phase of the moon" as the reason, or perhaps sunspots.

File: lib/git-version-bump.rb
Comment (p64) [reply lib/git-version-bump.rb:232]: I'll give it a whirl.

The pros of "run_command2" are I wouldn't have spent so much time thinking about a method that we throw out anyways. :)   At first to see if it worked I just hard coded system() instead of the run_command call. :)  Made me feel too dirty to even submit.

Comment (p64) [reply lib/git-version-bump.rb:232]: I'll give it a whirl.

**Comment by p64** [comment]
The pros of "run_command2" are I wouldn't have spent so much time thinking about a method that we throw out anyways. :) At first to see if it worked I just hard coded system() instead of the run_command call. :) Made me feel too dirty to even submit.

File: lib/git-version-bump.rb
Comment (p64) [reply lib/git-version-bump.rb]: Me and GUIs are not usually friends.  I think it says I've offered the commit.  Can you validate?  The only thing I was thinking as I wrote it is maybe we should be checking for true, false and else (unsupported value).  But that didn't seem to be consistent with what you wanted.  How's this look?

**Comment by p64** [comment]
I was just thinking, maybe people use GUIs?  So if one ran this from a CLI and it forked some GUI editor that popped a gui window, maybe that would have worked prior?  Not sure how people use it.

**Comment by mpalmer** [comment]
You're probably guessing correctly that many people have a GUI editor pop up for their commits, and that wouldn't suffer from the output-eating problem.  However, that can't be the whole answer, because I use a text mode editor (either `vim` or `joe`, depending on which machine I'm on) and as you might imagine, I've not hit this bug myself.  The Universe contains a multitude of mysteries, and I suspect this will go down in history as one of them.

**Review by mpalmer — APPROVED** [review]

**Comment by p64** [comment]
Does still seem like some sort of mystery.

Thank you for cutting a new release.  Much appreciated.

**Comment by p64** [comment]
I just pulled it.  Works like it should!

**Pull request merged**
